### PR TITLE
#8371: Documentation export as pdf

### PR DESCRIPTION
**Pull request by dsuren1**
## Description
This PR adds support for documentation export as PDF

**Changes done**
- Modified hyperlinks such that it works in both html and exported PDF file
- Replaced mermaid diagrams with png as mermaid diagrams cannot be exported to PDF (no plugin support)
- Documentation is downloaded with default name `mapstore_documentation.pdf`
- Added new plugin `mkdocs-pdf-export-plugin` (forked with fix for relatives hrefs)
- Dependency libraries added `Markdown` and `WeasyPrint`
- Exported document has combined pages of all md files

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Feature

## Issue

**What is the current behavior?**
#8371 

**What is the new behavior?**
Document can be exported as PDF via a small export icon on top of all the pages
image

_Sample can be found here_: https://mapstore2-temp.readthedocs.io/en/latest/

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Review by MV88 — CHANGES_REQUESTED** [review]
- [ ] i don't know why it is building here since locally i have some problems like
image


- [ ] links in docs/index.md fails in exported pdf
- [ ] syntax is wrong see
image
- [ ] can you add or improve needed changes in order to build the docs locally, maybe we can improve it [here](https://mapstore.readthedocs.io/en/latest/developer-guide/documentation-guidelines/)?

**Comment by MV88** [comment]
@dsuren1 I suggest also to use https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint to check syntax of the markdown

**Comment by dsuren1** [comment]
> * [ ]  i don't know why it is building here since locally i have some problems like
> * [ ]  links in docs/index.md fails in exported pdf

I don't see this error and also if you use the built docs from this PR https://mapstore--8455.org.readthedocs.build/en/8455/. Downloading PDF has index.md page, and is exported to PDF and the link in home.md `Managing Users and Groups` section works fine.

This is the pdf downloaded from the doc built in the [PR ](https://mapstore--8455.org.readthedocs.build/en/8455/pdf/mapstore_documentation.pdf)

> * [ ]  syntax is wrong see

I have only modified the href that has reference in another section which doesn't work in the exported pdf. So I haven't touched other areas of md files.

> * [ ]  can you add or improve needed changes in order to build the docs locally, maybe we can improve it [here](https://mapstore.readthedocs.io/en/latest/developer-guide/documentation-guidelines/)?

Sure, I can add some guidelines

**Comment by MV88** [comment]
> I will modify this. But in my defense, I have only modified the href that has reference in another section of md file which doesn't work in the exported pdf, so I left the other ones untouched.

we can open a dedicated issue for this, please ignore

**Comment by dsuren1** [comment]
> > I will modify this. But in my defense, I have only modified the href that has reference in another section of md file which doesn't work in the exported pdf, so I left the other ones untouched.
> 
> we can open a dedicated issue for this, please ignore

Sure. Thanks. But I have updated this alone for now.

**Comment by dsuren1** [comment]
@MV88 
I have fixed all the broken internal links and also update documentation guideline accordingly with a note.

**Comment by tdipisa** [comment]
@MV88 can you please complete the review?

**Comment by MV88** [comment]
@dsuren1 pr is conflicting please resolve it when you can

**Review by MV88 — CHANGES_REQUESTED** [review]
@dsuren1 
please check some of the comments
there some issues in files you have not touched

- [x] can you check and fix customize-theme.md which has a link to mapstore.geo-solutions.it?
- [x] in layers-settings the link of WMS Catalog Settings is not working
- [x] these two layer settings link points to a different url 
image
- [x] link of npm and node and python are not working in Requirements.md
- [x] database setup in Quick Build and Deploy is not working
- [x] More information on frontend building tools and configuration is available here link not working
- [x] some links inside Understanding frontend building tools not working, in particular
  - [x] karma.conf.continuous-test.js
  - [x] karma.conf.single-run.js
  - [x] tests.webpack.js
  - [ ] webpack.config.js
  - [ ] prod-webpack.config.js
- [ ] How to use a CDN link not working as it seems not to be a link, note the mouse pointer
image
- [ ] proxy.properties : configuration for the internal proxy (for cross-origin
requests). More information here, link not working

File: docs/developer-guide/documentation-guidelines.md [hunk 1]
```diff
@@ -4,4 +4,30 @@ Each new feature/tool in MapStore should be documented in the [User Guide](https
...
+   1. Install [**Python 3**](https://www.python.org/downloads/)
+   2. **Pip** is automatically installed when python is downloaded from python.org, if not, follow this instruction to install [pip](https://pip.pypa.io/en/stable/installation/)
+
+2. **Libraries installation**
+
+    Install **all** the libraries/plugins in `docs/requirements.txt` using **_pip_** while matching the exact version present
+
+    *Example*
+
+    For `mkdocs-material==3.2.0`, the installation using pip is as follows `pip install mkdocs-material==3.2.0`
```
[the remark this answers is not in the document]
Comment (dsuren1) [reply docs/developer-guide/documentation-guidelines.md:25]: pip installation of specific package version is denoted with `==`

[the remark this answers is not in the document]
Comment (MV88) [reply docs/developer-guide/documentation-guidelines.md:25]: oh really? I thought it was a typo 😃

**Review by MV88 — APPROVED** [review]
LGTM I think we are good now

**Comment by MV88** [comment]
@ElenaGallo testable in DEV

**Comment by ElenaGallo** [comment]
Test passed @dsuren1 please backport to the stable branch.

**Pull request merged**